publish-wrapper

```diff
@@ -0,0 +1,205 @@
+#!/usr/bin/env python3
+
+# This file is part of Cockpit.
+#
+# Copyright (C) 2019 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+import argparse
+import os
+import socket
+import subprocess
+import sys
+
+from task import github
+from task import sink
+
+sys.dont_write_bytecode = True
+
+HOSTNAME = socket.gethostname().split(".")[0]
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Log and publish into github')
+    parser.add_argument('--publish', help='Publish results centrally to a sink',
+                        default=os.environ.get("TEST_PUBLISH"))
+    parser.add_argument('--test-name', help='Test name',
+                        default=os.environ.get("TEST_NAME", "tests"))
+    parser.add_argument('--test-project', help="Project which is being tested",
+                        default=os.environ.get("TEST_PROJECT", ""))
+    parser.add_argument('--base-project', help="Project in which the tested PR is opened",
+                        default=os.environ.get("BASE_PROJECT"))
+    parser.add_argument('--test-branch', help="Branch to be tested - if differs to PR base branch",
+                        default=os.environ.get("BASE_BRANCH", ""))
```

Comment:
I still am not sure what would be the best approach, if to have only ENV or only positional arguments...

Reply:
My gut feeling is that in order to be meaningful the sink needs a certain set of properties, like `TEST_NAME, `TEST_PROJECT`. These should be positional parameters or mandatory options. Things like `BASE_BRANCH` might be optional, and thus should become proper options.

`TEST_PUBLISH` now seems kind of meaningless to publish-wrapper itself; run-queue should look at this and call publish-wrapper around the command if it is set. However, I realize that this might be difficult as run-queue shouldn't interpret the command (and read TEST_PROJECT etc.) In tests-scan it's also too early, as that runs on a different environment (webhook) that doesn't have `$TEST_PUBLISH`. So maybe leave the `--publish` option as-is for now?